src/__tests__/Events.test.tsx

```diff
@@ -0,0 +1,384 @@
+        rotation: 3.14,
+      },
+      { numberOfPointers: 2 },
+    ],
+  ])(
+    'receives additional properties depending on handler type ("%s")',
+    (
+      handlerName: string,
+      additionalEventData: Record<string, unknown>,
+      defaultEventData: Record<string, unknown>
+    ) => {
+      const handlers = mockedEventHandlers();
+      const { getByTestId } = render(
+        <NestedHandlers eventHandlers={handlers} />
+      );
+
+      fireGestureHandler(getByTestId(handlerName), [
+        {
+          ...additionalEventData,
+          oldState: State.UNDETERMINED,
+          state: State.BEGAN,
+        },
+        {
+          ...additionalEventData,
+          oldState: State.BEGAN,
+          state: State.ACTIVE,
+        },
+      ]);
+
+      expect(handlers.begin).lastCalledWith({
+        nativeEvent: expect.objectContaining({
+          ...additionalEventData,
+          ...defaultEventData,
+        }),
+      });
+    }
+  );
+
+  it('fills oldState if not passed', () => {
+    const handlers = mockedEventHandlers();
+    const { getByTestId } = render(<SingleHandler eventHandlers={handlers} />);
+    fireGestureHandler(getByTestId('pan'), [
+      { state: State.BEGAN },
+      { state: State.ACTIVE },
+      { state: State.ACTIVE },
+      { state: State.ACTIVE },
+      { state: State.END },
+    ]);
+
+    expect(handlers.begin).toBeCalledWith({
+      nativeEvent: expect.objectContaining({ oldState: State.UNDETERMINED }),
+    });
+    expect(handlers.active).nthCalledWith(1, {
+      nativeEvent: expect.objectContaining({ oldState: State.BEGAN }),
+    });
+    expect(handlers.active).lastCalledWith({
+      nativeEvent: expect.not.objectContaining({ oldState: expect.anything() }),
+    });
+    expect(handlers.end).toBeCalledWith({
+      nativeEvent: expect.objectContaining({ oldState: State.ACTIVE }),
+    });
+  });
+});
+
+describe('Using Reanimated 2 useAnimatedGestureHandler hook', () => {
+  function UseAnimatedGestureHandler({ eventHandlers }: V1ApiProps) {
+    const eventHandler = useAnimatedGestureHandler({
+      onStart: eventHandlers.begin,
+    });
+    return (
+      <LongPressGestureHandler
+        testID="longPress"
+        onHandlerStateChange={eventHandler}>
+        <Text>Long press handler</Text>
+      </LongPressGestureHandler>
+    );
+  }
+
+  it('calls callback with (event data, context)', () => {
+    const handlers = mockedEventHandlers();
+    const { getByTestId } = render(
+      <UseAnimatedGestureHandler eventHandlers={handlers} />
+    );
+
+    fireGestureHandler(getByTestId('longPress'), [
+      { state: State.BEGAN },
+      { state: State.ACTIVE },
+      { state: State.END },
+    ]);
+
+    expect(handlers.begin).toBeCalledWith(
+      expect.objectContaining({ state: State.BEGAN }),
+      expect.any(Object)
+    );
+  });
+});
+
+describe('Using RNGH v2 gesture API', () => {
+  function SingleHandler({ handlers, treatStartAsUpdate }: any) {
```

Comment:
I think we can create a type for v2 similarly to how it looks for v1:
```ts
interface V2ApiProps {
  handlers: ReturnType<typeof mockedEventHandlers>;
  treatStartAsUpdate?: boolean;
}
```
Some functions will be unnecessary but `mockedEventHandlers` is used to generate handlers anyway.

Reply:
Yes, I just forgot to type it properly.

> Some functions will be unnecessary but mockedEventHandlers is used to generate handlers anyway.

I don't understand this statement, could you expand on that?

Reply:
`fail` and `cancel` can't be 1:1 mapped to the new API, they trigger `onEnd` with `false` as the second argument.

Reply:
Fixed in 0b089c0c.